include/ensmallen_bits/moead/weight_init_policies/dirichlet_init.hpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file dirichlet_init.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * The Dirichlet method of Weight Initialization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_MOEAD_DIRICHLET_HPP
+#define ENSMALLEN_MOEAD_DIRICHLET_HPP
+
+namespace ens {
+
+/**
+ * The Dirichlet method for initializing weights. Sampling a dirichlet distribution
+ * with parameters set to one returns points lying on unit simplex with uniform
+ * distribution.
+ *
+ */
+class Dirichlet
+{
+ public:
+  /**
+   * Constructor for Dirichlet policy.
+   */
+  Dirichlet()
+  {
+    /* Nothing to do. */
+  }
+
+  /**
+   * Generate the reference direction matrix.
+   *
+   * @tparam MatType The type of the matrix used for constructing weights.
+   * @param numObjectives The dimensionality of objective space.
+   * @param numPoints The number of reference directions requested.
+   * @param epsilon Handle numerical stability after weight initialization.
+   */
+  template<typename MatType>
+  MatType Generate(const size_t numObjectives,
+                   const size_t numPoints,
+                   const double epsilon)
+  {
+    MatType weights = arma::randg<MatType>(numObjectives, numPoints,
+        arma::distr_param(1.0, 1.0)) + epsilon;
+    // Normalize each column.
+    return arma::normalise(weights, 1, 0);
```

Comment:
So, thinking about this... We have a gamma distribution given as: 

```
Y_i ~ gamma(alpha_i, 1)
```

From there, we're creating the normalization of:

```
X = sum_{i = 1}^{n} Y_i
V = (Y_1/X, Y_2/X, ... Y_n/X) ~ Dirchlet(alpha_1, ... alpha_n)
```
https://en.wikipedia.org/wiki/Dirichlet_distribution#Gamma_distribution

Looking at: 

http://arma.sourceforge.net/docs.html#normalise

`normalise` is giving is a unit _p_-norm. With `p = 1`, we're getting: 

```
X = sum abs(Y_i)
```

The `abs()` goes against the summation component, but gamma RVs will always be > 0.

So, long story short... We're doing one extra operation, but the end outcome will be the same.

Reply:
I think ```p=1``` gives something like

```std::sqrt( elem0^2 + elem1^2 + elem2^2 ..... elemn^2)``` and not the sum if I've gotten this correctly. I suppose you're saying, assuming X = sum abs(Y_i); abs part is unnecessary?

Reply:
Nope, that would be `p=2`/ l2-norm. This is using an l1-norm by column.

`p=1` gives: 

```cpp
std::pow( abs(elem0)^1 + abs(elem1)^1 + abs(elem2)^1 ..... abs(elemn)^1, 1.0);
```
or:

```
(abs(elem0)^1 + abs(elem1^1 + abs(elem2^1  + ..... + abs(elemn^1))^1 
= abs(elem0)^1 + abs(elem1)^1 + abs(elem2)^1  + ..... +  abs(elemn)^1 
= abs(elem0) + abs(elem1) + abs(elem2) + ..... + abs(elemn)
```

 I think by-column is correct instead of by-row due to the weights being structured `numObjectives x numPoints` (norm across objectives under a single point)

Reply:
Yes, I've done it in a column basis so the normalization of each point is ensured. Just revised L-norms, I was mistaken about the L1 norm formula, thanks for pointing it out.

I think the "abs" part shouldn't really be that of a concern, either way, we know each element is > 0. We have one line of code which do the thing for us, and I really don't think the additional "abs" would severely hinder the performance.